src/state/farms/elastic/hooks.ts

```diff
@@ -0,0 +1,769 @@
+      position: {
+        id: string
+        liquidity: string
+        tickLower: {
+          tickIdx: string
+        }
+        tickUpper: {
+          tickIdx: string
+        }
+      }
+    }>
+    pool: {
+      id: string
+      feeTier: string
+      sqrtPrice: string
+      feesUSD: string
+      liquidity: string
+      tick: string
+      totalValueLockedUSD: string
+      reinvestL: string
+      token0: SubgraphToken
+      token1: SubgraphToken
+    }
+  }>
+}
+
+const POOL_FEE_HISTORY = gql`
+  query poolFees($block: Int!, $poolIds: [String]!) {
+    pools(block: { number: $block }, where: { id_in: $poolIds }) {
+      id
+      feesUSD
+    }
+  }
+`
+
+const ELASTIC_FARM_QUERY = gql`
+  query getFarms {
+    farms(first: 1000) {
+      id
+      rewardLocker
+      farmingPools {
+        id
+        pid
+        startTime
+        endTime
+        feeTarget
+        vestingDuration
+        rewardTokens {
+          token {
+            id
+            symbol
+            name
+            decimals
+          }
+          amount
+        }
+        joinedPositions {
+          id
+          user
+          pid
+          liquidity
+          position {
+            id
+            liquidity
+            tickLower {
+              tickIdx
+            }
+            tickUpper {
+              tickIdx
+            }
+          }
+        }
+        pool {
+          id
+          feeTier
+          tick
+          totalValueLockedUSD
+          liquidity
+          feesUSD
+          reinvestL
+          sqrtPrice
+          token0 {
+            id
+            symbol
+            name
+            decimals
+          }
+          token1 {
+            id
+            symbol
+            name
+            decimals
+          }
+        }
+      }
+    }
+  }
+`
+
+const farmInterface = new Interface(ELASTIC_FARM_ABI)
+const positionManagerInterface = new Interface(NFTPositionManagerABI.abi)
+
+export const useElasticFarms = () => {
+  const { chainId } = useActiveWeb3React()
+  const defaultChainData = useMemo(
+    () => ({
+      loading: false,
+      farms: null,
+      poolFeeLast24h: {},
+    }),
+    [],
+  )
+  return useAppSelector(state => state.elasticFarm)[chainId || 1] || defaultChainData
+}
+
+export const FarmUpdater = ({ interval = true }: { interval?: boolean }) => {
+  const dispatch = useAppDispatch()
+  const { chainId, account } = useActiveWeb3React()
+  const defaultChainData = useMemo(
+    () => ({
+      loading: false,
+      farms: null,
+      poolFeeLast24h: {},
+    }),
+    [],
+  )
+  const elasticFarm = useAppSelector(state => state.elasticFarm)[chainId || 1] || defaultChainData
+
+  const [getElasticFarms, { data, error }] = useLazyQuery(ELASTIC_FARM_QUERY, {
+    client: NETWORKS_INFO[chainId || ChainId.MAINNET].elasticClient,
+    fetchPolicy: 'network-only',
+  })
+
+  useEffect(() => {
+    if (!elasticFarm.farms && !elasticFarm.loading) {
+      console.time('getFarmFromSubgraph')
```

Comment:
```suggestion
```

Reply:
console.time for performance checking on production. will remove later on